src/components/application_manager/test/mobile_message_handler_test.cc

```diff
@@ -233,13 +233,15 @@ class MobileMessageHandlerTest : public testing::Test {
                                                             data,
                                                             bin_dat);
     // Act
-    RawMessage* result_message =
-        MobileMessageHandler::HandleOutgoingMessageProtocol(message_to_send);
+    std::shared_ptr<RawMessage> result_message(
+        MobileMessageHandler::HandleOutgoingMessageProtocol(message_to_send));
     std::vector<uint8_t> full_data = joiner<std::vector<uint8_t> >(
         binary_header, binary_header + PROTOCOL_HEADER_V2_SIZE, data);
     size_t full_size =
         sizeof(uint8_t) * full_data.size() + bin_dat->size() * sizeof(uint8_t);
 
+    delete bin_dat;
```

Comment:
@SNiukalov Why do you use heap? Is it possible to use stack for bin_dat? ?

Reply:
@AByzhynar 
Fixed in:https://github.com/LuxoftSDL/sdl_core/pull/41/commits/be4a738c2392d04c4fd0cd9a2dee39ce76b94c6e